### ISSUE TITLE
metadata property eo:cloud_cover == ... not immediately rejected

### ISSUE THREAD

**Issue by bossie** [Open-EO/openeo-geotrellis-extensions#39]
Sentinel Hub doesn't support filtering on exact values for "eo:cloud_cover" ("eq"), only on a maximum value ("lte"). Apparently this one was able to slip by where it should have been rejected immediately:

```python
 "eo:cloud_cover": lambda v: v == 80
```

**Comment by jdries** [comment]
@soxofaan

**Comment by bossie** [comment]
untested:

```scala
case (metadataProperty, "eq", value) if metadataProperty != "eo:cloud_cover" => metadataProperty -> value
```

at https://github.com/Open-EO/openeo-geotrellis-extensions/blob/3b1a68a396e1b07e0f61323bbc4ff134e9d99a78/geotrellis-sentinelhub/src/main/scala/org/openeo/geotrellissentinelhub/Criteria.scala#L47

**Comment by jdries** [comment]
we also need to check if 'lte' works well with terrascope layer, so we can move to using that consistently

**Comment by bossie** [comment]
Looks like the original problem in this issue was fixed (but not properly linked) in 0901530b932083d1b36d8129b28cad13b2a7c589.
